packages/flutter/test/material/text_field_test.dart

```diff
@@ -3341,4 +3341,38 @@ void main() {
     expect(minOffset, 0.0);
     expect(maxOffset, 50.0);
   });
+
+  testWidgets('TextField style is merged with theme', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/23994
+
+    final ThemeData themeData = ThemeData(
+      textTheme: TextTheme(
+        subhead: TextStyle(
+          color: Colors.blue[500],
+        ),
+      ),
+    );
+
+    Widget buildFrame() {
```

Comment:
It would be good to test this version and one where the TextStyle overrides many properties and then one where the TextField's textStyle has inherit:true, which is a special case for merge: see https://docs.flutter.io/flutter/painting/TextStyle/merge.html